rubix-core/src/test/java/com/qubole/rubix/core/TestUpdatingofCache.java

```diff
@@ -0,0 +1,206 @@
+        conf.setBoolean(CachingConfigHelper.DATA_CACHE_STRICT_MODE, true);
+        conf.setInt(BookKeeperConfig.DATA_CACHE_BOOKKEEPER_PORT, 3456);
+        Thread thread = new Thread() {
+            public void run()
+            {
+                BookKeeperServer.startServer(conf);
+            }
+        };
+        thread.start();
+
+        DataGen.populateFile(backendFileName);
+        LocalFSInputStream localFSInputStream = new LocalFSInputStream(backendFileName);
+        FSDataInputStream fsDataInputStream = new FSDataInputStream(localFSInputStream);
+
+        File file = new File(backendFileName);
+        conf.setInt(BookKeeperConfig.BLOCK_SIZE, blockSize);
+
+        while (!BookKeeperServer.isServerUp()) {
+            Thread.sleep(200);
+            log.info("Waiting for BookKeeper Server to come up");
+        }
+        log.info("All set to test");
+
+        // This should be after server comes up else client could not be created
+        inputStream = new CachingInputStream(fsDataInputStream, conf, backendPath, file.length(),file.lastModified(), new CachingFileSystemStats());
+    }
+
+    public void setup()
+            throws InterruptedException, IOException
+    {
+        final Configuration conf = new Configuration();
+        conf.setBoolean(CachingConfigHelper.DATA_CACHE_STRICT_MODE, true);
+        conf.setInt(BookKeeperConfig.DATA_CACHE_BOOKKEEPER_PORT, 3456);
+
+        File file = new File(backendFileName);
+
+        LocalFSInputStream localFSInputStream = new LocalFSInputStream(backendFileName);
+        FSDataInputStream fsDataInputStream = new FSDataInputStream(localFSInputStream);
+        conf.setInt(BookKeeperConfig.BLOCK_SIZE, blockSize);
+
+        log.info("All set to test 1");
+
+        // This should be after server comes up else client could not be created
+        inputStream = new CachingInputStream(fsDataInputStream, conf, backendPath, file.length(),file.lastModified(), new CachingFileSystemStats());
+
+
+    }
+
+    @AfterMethod
+    public void cleanup()
+    {
+        BookKeeperServer.stopServer();
+        Configuration conf = new Configuration();
+        inputStream.close();
+        File file = new File(backendFileName);
+        file.delete();
+
+        File mdFile = new File(BookKeeperConfig.getMDFile(backendPath.toString(), conf));
+        mdFile.delete();
+
+        File localFile = new File(BookKeeperConfig.getLocalPath(backendPath.toString(), conf));
+        localFile.delete();
+    }
+
+    private void testEvictionHelper()
+            throws IOException
+    {
+        inputStream.seek(100);
+        byte[] buffer = new byte[1000];
+        int readSize = inputStream.read(buffer, 0, 1000);
+        String expectedOutput = DataGen.generateContent().substring(100, 1100);
+        assertions(readSize, 1000, buffer,expectedOutput);
+    }
+
+    @Test
+    public void testEviction()
+            throws IOException, InterruptedException
```

Comment:
There is lot of code duplication here with only this method (testEviction) being new code. Move this test to TestCachingInputStream, that will save lot of duplication